doc/content/reference/id-eui-constraints/index.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "ID and EUI Constraints"
+description: ""
+---
+
+This reference explains how {{% tts %}} deals with IDs and EUIs, including regular expression requirements, uniqueness, and how deletion is handled.
+
+<!--more-->
+
+## Requirements of an ID or EUI
+
+An ID or EUI in {{% tts %}} must:
+
+- Have a length of 3-36 characters
+- Consist of lowercase letters, numbers, and non-consecutive dashes
+
+The following regular expression is used to validate IDs and EUIs:
+
+`(^[a-z0-9](?:[-]?[a-z0-9]){2,}$)`
+
+## Uniqueness
+
+Within a deployment, there can only be:
+
+- One tenant with the same ID
+- One gateway with the same EUI (across all tenants)
+- One end device with the same DevEUI (across all tenants)
```

Comment:
This is not true. A DevEUI may be used in two tenants.

Reply:
took this from @htdvisser comment https://github.com/TheThingsIndustries/lorawan-stack-docs/issues/211#issue-795879347.

mistake?

Reply:
According to the definition of a Dev EUI, it is a globally unique identifier, so across all tenants.

The Identity Server only enforces uniqueness of the (JoinEUI,DevEUI) tuple across all tenants. For the record: I think we should be more strict, but this is what we decided to do.

Reply:
Updated, awaiting review @adriansmares